share/functions/__fish_cache_read.fish

```diff
@@ -0,0 +1,10 @@
+function __fish_cache_read -a cache_file max_age
+    if not test -f $cache_file
+        return 1
+    end
+    set -l age (path mtime --relative $cache_file)
+    if test -n "$max_age" && test $age -gt $max_age
+        return 1
+    end
+    cat $cache_file
+end
```

Comment:
this function call is always followed by a call to `__fish_cache_put`,
Can we inline that one into `__fish_cache_read`?

For extra points, we could also teach this function how to populate the cache;
we could pass a "callback" argument that we do "eval $callback" on.
Then we should only pass the basename of the cache file to this function (and have it call __fish_make_cache_dir itself).
Then a good name for the function might be

	__fish_cached <cache-key> <max-age> 'echo <contents>'

Reply:
Working on that now. 

Instead of being a parameter, the cache key could be created based on the command name, something like that:

`__fish_cached -t 250 -- pacman -Ssq \| sed -e 's/\$/\tPackage/'`

Extracting the first part of  the command, we get `pacman`, that can be used as the cache key.

I'm wondering if the functions `__fish_cache_read` and `__fish_cache_put` should be keep as separate functions or should be merged inside the `__fish_cached` function. @krobelus  What do you think?

Reply:
An optional argument could also be provided to explicit specify a cache key